pandas/tests/io/formats/test_format.py

```diff
@@ -3167,6 +3166,65 @@ def test_str(self):
         assert str(NaT) == "NaT"
 
 
+class TestPeriodIndexFormat:
+    def test_period(self):
+        """Basic test for period formatting with default format."""
+        p = pd.PeriodIndex([datetime(2003, 1, 1, 12), None], freq="H")
+        # default formatting
+        formatted = p.format()
+        assert formatted[0] == "2003-01-01 12:00"  # default: minutes not shown
+        assert formatted[1] == "NaT"
+        # format is equivalent to strftime(None)...
+        assert formatted[0] == p.strftime(None)[0]
+        assert p.strftime(None)[1] is np.nan  # ...except for NaTs
+
+        # Same test with nanoseconds freq
+        p = pd.period_range("2003-01-01 12:01:01.123456789", periods=2, freq="n")
+        formatted = p.format()
+        assert (formatted == p.strftime(None)).all()
+        assert formatted[0] == "2003-01-01 12:01:01.123456789"
+        assert formatted[1] == "2003-01-01 12:01:01.123456790"
+
+    def test_period_custom(self):
```

Comment:
Can this be parameterized?

Reply:
indeed

Reply:
An alternative (resulting in the exact same test as current, not a weaker one) is to have a lot of parameters:

```python
@pytest.mark.parametrize("freq,float_str,ms,us,ns,ms2,us2,ns2", [
    ("l", "123", "123", "123000", "123000000", "124", "124000", "124000000"),
    ("u", "123456", "123", "123456", "123456000", "123", "123457", "123457000"),
    ("n", "123456789", "123", "123456", "123456789", "123", "123456", "123456790"),
])
def test_period_custom(self, freq, float_str, ms, us, ns, ms2, us2, ns2):
    # GH#46252 custom formatting directives %l (ms) and %u (us)

    per = pd.period_range(f"2003-01-01 12:01:01.{float_str}", periods=2, freq=freq)
    formatted = per.format(date_format=f"%y %I:%M:%S (ms=%l us=%u ns=%n)")
    assert formatted[0] == f"03 12:01:01 (ms={ms} us={us} ns={ns})"
    assert formatted[1] == f"03 12:01:01 (ms={ms2} us={us2} ns={ns2})"
```

This is quite less readable in my opinion.
If you like I can include this, let me know.

Reply:
Okay nevermind, yeah the existing test looks okay then.